### ISSUE TITLE
IDE0004 false positive when we need to box a nullable value type, and we have a cast in place.

### ISSUE THREAD

**Issue by zlatanov** [dotnet/roslyn#65005]
**Version Used**: 
Visual Studio 2022, 17.3.6

**Steps to Reproduce**:
```c#
byte? temp = 10;
object? box = (int?)temp;

Console.WriteLine((int?)box);
```

The cast of the `temp` variable to `int?` before boxing occurs is incorrectly flagged as unnecessary. If we remove it, we will be unable to unbox the value later as int.
